protocol/slp/slp.md

```diff
@@ -0,0 +1,78 @@
+The GENESIS message defines the SLP token and issues the first batch of tokens.
+The MINT message issues further batches of tokens.
+The SEND message denotes the number of tokens sent to each output.
+The COMMIT message is proposed to allow periodical commits of all transaction hashes on the token.
+
+
+## Consensus rules
+
+In all cases of SLP transactions, there must be an [output script](/protocol/blockchain/transaction/unlocking-script) that begins with OP_RETURN (opcode 0x6a) within the first output (vout=0).
+The data output should be an SLP message that conforms to the predefined format.
+
+
+For transfer transactions (SEND), the sum of token outputs specified OP_RETURN may not exceed the sum of valid token inputs.
+Valid token inputs are transaction inputs where the previous transactions are known to conform to SLP consensus rules.
+The values of valid token inputs are as listed in SLP messages of the transactions that create the transaction outputs.
+Inputs of differing token types or token ids are ignored.
+
+For issuance transactions (GENESIS/MINT), Genesis transactions are self-evidently valid or invalid, and MINT transactions require the baton input for the token being issued.
+The baton is either directly from the Genesis transaction that created it, or indirectly via the previous Mint transaction.
+Inputs of differing token types or token ids are ignored.
+Token inputs are also ignored.
+
+
+## Validation of the SLP transactions
+
+The “colored coins” design ensures that any transaction that attempts to double-spend SLP token outputs will also double-spend the BCH outputs, and thus, will be rejected by the network.
+However, invalid SLP token transactions will be accepted by the network and included in the Blockchain if they are contained within valid BCH transactions.
+Therefore, validation of an SLP transaction requires a directed acyclic graph (DAG), with transactions as its vertices and transaction inputs as its edges, that originates from the transaction that is being validated and terminates at the Mint or Genesis transaction.
+Full nodes can construct the DAG proof for an SLP transaction by recursively iterating through the inputs of the said transaction to trace their ancestor transactions.
+
+
+Several optimizations have been made to this process.
+Vertices of transactions that provide zero tokens can be pruned, so are the input edges of known valid transactions.
+Otherwise, DAG proof for SLP transactions can be cached to accelerate the proof of future SLP transactions if pruning is not employed.
+The process can also stop before the full DAG is constructed when the sum of proved valid inputs exceeds the output sum, or when the sum of possibly valid inputs goes below the output sum.
+
+Without DAG proofs, other validation methods are also available.
+Users can trust SLP transactions with incomplete DAG proofs except for when the attacker creates longer chains than the DAG proofs.
```

Comment:
I added some explanations here in the following commits. However, I will leave this conversation here just in case if you have anything more to add about the explanations. This is a little bit tricky because I don't want the explanation to be either wordy or insufficient.